### PR TITLE
ISPN-10855 Remove the need for factory() implementations when the base class has it

### PR DESCRIPTION
**Pull request by danberindei**
https://issues.jboss.org/browse/ISPN-10855

### CODE REVIEW

**Comment by anistor** [comment]
Checkstyle violations, @danberindei . This did not run.

File: integrationtests/elasticsearch-indexmanager-it/src/test/java/org/infinispan/query/it/ClusteredCacheWithElasticsearchIndexManagerIT.java [hunk 1]
```diff
@@ -17,6 +18,10 @@
  */
 @Test(groups = "functional", testName = "query.blackbox.ClusteredCacheWithElasticsearchIndexManagerIT")
 public class ClusteredCacheWithElasticsearchIndexManagerIT extends ClusteredCacheTest {
+    @Override
+    public Object[] factory() {
```
Comment (anistor) [inline integrationtests/elasticsearch-indexmanager-it/src/test/java/org/infinispan/query/it/ClusteredCacheWithElasticsearchIndexManagerIT.java]: The fix makes me sad. But such is life after you've adopted TestNG.

Comment (danberindei) [reply integrationtests/elasticsearch-indexmanager-it/src/test/java/org/infinispan/query/it/ClusteredCacheWithElasticsearchIndexManagerIT.java]: We make the decision whether to run the factory method or not in `MultipleCacheManagersTest`. I could have changed it to never throw an exception and just ignore the `factory()` method from super-classes, if that makes you feel better @anistor

**Comment by danberindei** [comment]
Updated to ignore the `factory()` method inherited from a super class instead of failing the test.

The original intention was to fail the test, but before #7498 the simulated test failure did not work and it was effectively ignoring the `factory()` method. So this is reverting to the pre- #7498 behaviour, only making it explicit.

**Comment by gustavonalle** [comment]
I guess master is currently broken right? I just rebased my PR and started getting tons of
```
org.infinispan.query.blackbox.ClusteredCacheWithInfinispanDirectoryTest.factory() override is missing, inherited factory() creates instances of org.infinispan.query.blackbox.ClusteredCacheTest
```

**Comment by danberindei** [comment]
@gustavonalle yes, this PR fixes the errors you're seeing.

**Pull request merged**